rclpy/test/test_client.py

```diff
@@ -234,11 +234,11 @@ def _service(request, response):
             executor_thread = threading.Thread(
                 target=TestClient._spin_rclpy_node, args=(self.node, executor))
             executor_thread.start()
-            result = cli.call(GetParameters.Request(), 0.5)
-            self.assertTrue(result is None)
+            with self.assertRaises(TimeoutError):
+                cli.call(GetParameters.Request(), 0.5)
+        finally:
             executor.shutdown()
             executor_thread.join()
```

Comment:
You have to be careful here, because if you try to join a thread that wasn't started it will throw again here:

https://docs.python.org/3/library/threading.html#threading.Thread.join

>[join()](https://docs.python.org/3/library/threading.html#threading.Thread.join) raises a [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError) if an attempt is made to join the current thread as that would cause a deadlock. It is also an error to [join()](https://docs.python.org/3/library/threading.html#threading.Thread.join) a thread before it has been started and attempts to do so raise the same exception.

Reply:
thanks, `is_alive()` is added to this particular test file to make sure the thread has been started once. i can see some other places where have similar test codes like here, but i think that would be better to address with another PR.